src/ResourceManager/KeyVault/Commands.KeyVault/Models/DataServiceCredential.cs

```diff
@@ -74,21 +74,23 @@ public string GetToken()
 
         private static string GetTenantId(IAzureContext context)
         {
-            var tenantId = string.Empty;
             if (context.Account == null)
+            {
                 throw new ArgumentException(KeyVaultProperties.Resources.ArmAccountNotFound);
+            }
 
-            if (context.Account.Type != AzureAccount.AccountType.User &&
-                context.Account.Type != AzureAccount.AccountType.ServicePrincipal)
-                throw new ArgumentException(string.Format(KeyVaultProperties.Resources.UnsupportedAccountType, context.Account.Type));
-
-            if (context.Subscription != null && context.Account != null)
+            var tenantId = string.Empty;
+            if (context.Tenant != null && context.Tenant.GetId() != Guid.Empty)
+            {
+                tenantId = context.Tenant.Id.ToString();
+            }
+            else if (string.IsNullOrWhiteSpace(tenantId) && context.Subscription != null && context.Account != null)
```

Comment:
I don't think you need to check if the tenantId is empty - you just set it to empty above